ProcessLib/HT/HTFEM.h

```diff
@@ -116,178 +79,82 @@ class LocalAssemblerData : public HTLocalAssemblerInterface
-                _process_data.thermal_conductivity_solid(t, pos)[0];
-            auto const thermal_conductivity_fluid =
-                _process_data.thermal_conductivity_fluid(t, pos)[0];
-
-            auto const& ip_data = _ip_data[ip];
-            auto const& N = ip_data.N;
-            auto const& dNdx = ip_data.dNdx;
-            auto const& w = ip_data.integration_weight;
-
-            double T_int_pt = 0.0;
-            double p_int_pt = 0.0;
-            // Order matters: First T, then P!
-            NumLib::shapeFunctionInterpolate(local_x, N, T_int_pt, p_int_pt);
-
-            // \todo the first argument has to be changed for non constant
-            // porosity model
-            auto const porosity =
-                _process_data.porous_media_properties.getPorosity(t, pos)
-                    .getValue(t, pos, 0.0, T_int_pt);
-
-            auto const intrinsic_permeability =
-                _process_data.porous_media_properties.getIntrinsicPermeability(
-                    t, pos).getValue(t, pos, 0.0, T_int_pt);
+        auto const& N = _ip_data[integration_point].N;
 
-            double const thermal_conductivity =
-                thermal_conductivity_solid * (1 - porosity) +
-                 thermal_conductivity_fluid * porosity;
+        // assumes N is stored contiguously in memory
+        return Eigen::Map<const Eigen::RowVectorXd>(N.data(), N.size());
+    }
 
-            auto const specific_heat_capacity_solid =
-                _process_data.specific_heat_capacity_solid(t, pos)[0];
+protected:
+    MeshLib::Element const& _element;
+    HTMaterialProperties const& _material_properties;
 
-            vars[static_cast<int>(
-                MaterialLib::Fluid::PropertyVariableType::T)] = T_int_pt;
-            vars[static_cast<int>(
-                MaterialLib::Fluid::PropertyVariableType::p)] = p_int_pt;
-            auto const specific_heat_capacity_fluid =
-                _process_data.fluid_properties->getValue(
-                    MaterialLib::Fluid::FluidPropertyType::HeatCapacity, vars);
+    IntegrationMethod const _integration_method;
+    std::vector<
+        IntegrationPointData<NodalRowVectorType, GlobalDimNodalMatrixType>,
+        Eigen::aligned_allocator<
+            IntegrationPointData<NodalRowVectorType, GlobalDimNodalMatrixType>>>
+        _ip_data;
 
-            auto const thermal_dispersivity_longitudinal =
-                _process_data.thermal_dispersivity_longitudinal(t, pos)[0];
-            auto const thermal_dispersivity_transversal =
-                _process_data.thermal_dispersivity_transversal(t, pos)[0];
+    double getHeatEnergyCoefficient(const double t, const SpatialPosition& pos,
+                                    const double porosity,
+                                    const double fluid_density,
+                                    const double specific_heat_capacity_fluid)
+    {
+        auto const& material_properties = this->_material_properties;
```

Comment:
Could you remind me please, why `this` is necessary here? I'd think it's not.

Reply:
Removed.